src/ast/nodes/AssignmentExpression.ts

```diff
@@ -52,29 +52,38 @@ export default class AssignmentExpression extends NodeBase {
 		this.left.render(code, options);
 		this.right.render(code, options);
 		if (options.format === 'system') {
-			if (this.left.variable && this.left.variable.exportName) {
+			const exportNames =
+				this.left.variable && options.exportNamesByVariable.get(this.left.variable);
+			if (this.left.type === 'Identifier' && exportNames) {
+				const _ = options.compact ? '' : ' ';
 				const operatorPos = findFirstOccurrenceOutsideComment(
 					code.original,
 					this.operator,
 					this.left.end
 				);
 				const operation =
-					this.operator.length > 1
-						? ` ${this.left.variable.exportName} ${this.operator.slice(0, -1)}`
-						: '';
+					this.operator.length > 1 ? `${exportNames[0]}${_}${this.operator.slice(0, -1)}` : '';
+				const nextIsSpace =
```

Comment:
Honestly I think this is a little over-the-top. I agree it makes the output a little nicer in the common case if there is a single space, but
- it adds quite a bit of complexity and makes it harder to understand the code
- we never do this in many other similar situations

So I would hope we can get rid of these micro-optimizations (`nextIsSpace` and `nextIsNewline`) in favour of easier code maintenance.